Test error in TestCmdEvalReply

On Fedora Rawhide with Golang 1.15.6:

```
--- FAIL: TestCmdEvalReply (0.01s)
    cmd_scripting_test.go:438: expected "string expected, got number" in "-1\r\n"
    cmd_scripting_test.go:448: expected "string expected, got number" in "+1\r\n"
```

--- a/cmd_scripting_test.go
+++ b/cmd_scripting_test.go
@@ -432,22 +432,22 @@ func TestCmdEvalReply(t *testing.T) {
 
 	mustContain(t, c,
 		"EVAL", `return redis.error_reply()`, "0",
-		"string expected, got nil",
+		"wrong number or type of arguments",
 	)
 
 	mustContain(t, c,
 		"EVAL", `return redis.error_reply(1)`, "0",
-		"string expected, got number",
+		"wrong number or type of arguments",
 	)
 
 	mustContain(t, c,
 		"EVAL", `return redis.status_reply()`, "0",
-		"string expected, got nil",
+		"wrong number or type of arguments",
 	)
 
 	mustContain(t, c,
 		"EVAL", `return redis.status_reply(1)`, "0",
-		"string expected, got number",
+		"wrong number or type of arguments",
 	)
 }
 

--- a/go.mod
+++ b/go.mod
@@ -2,7 +2,7 @@ module github.com/alicebob/miniredis/v2
 
 require (
 	github.com/alicebob/gopher-json v0.0.0-20200520072559-a9ecdc9d1d3a
-	github.com/yuin/gopher-lua v0.0.0-20191220021717-ab39c6098bdb
+	github.com/yuin/gopher-lua v0.0.0-20200816102855-ee81675732da
 )
 
 go 1.13

--- a/integration/script_test.go
+++ b/integration/script_test.go
@@ -119,12 +119,12 @@ func TestLua(t *testing.T) {
 		c.Do("EVAL", "return {ok = 'great', notok = 'yes'}", "0")       // doc error?
 		c.Do("EVAL", "return {1, 2, ok = 'great', notok = 'yes'}", "0") // doc error?
 
-		c.DoLoosely("EVAL", "return redis.error_reply(1)", "0")
-		c.DoLoosely("EVAL", "return redis.error_reply()", "0")
-		c.DoLoosely("EVAL", "return redis.error_reply(redis.error_reply('foo'))", "0")
-		c.DoLoosely("EVAL", "return redis.status_reply(1)", "0")
-		c.DoLoosely("EVAL", "return redis.status_reply()", "0")
-		c.DoLoosely("EVAL", "return redis.status_reply(redis.status_reply('foo'))", "0")
+		c.Error("type of arguments", "EVAL", "return redis.error_reply(1)", "0")
+		c.Error("type of arguments", "EVAL", "return redis.error_reply()", "0")
+		c.Error("type of arguments", "EVAL", "return redis.error_reply(redis.error_reply('foo'))", "0")
+		c.Error("type of arguments", "EVAL", "return redis.status_reply(1)", "0")
+		c.Error("type of arguments", "EVAL", "return redis.status_reply()", "0")
+		c.Error("type of arguments", "EVAL", "return redis.status_reply(redis.status_reply('foo'))", "0")
 	})
 
 	// state inside lua

--- a/lua.go
+++ b/lua.go
@@ -95,14 +95,24 @@ func mkLuaFuncs(srv *server.Server, c *server.Peer) map[string]lua.LGFunction {
 		"call":  mkCall(true),
 		"pcall": mkCall(false),
 		"error_reply": func(l *lua.LState) int {
-			msg := l.CheckString(1)
+			v := l.Get(1)
+			msg, ok := v.(lua.LString)
+			if !ok {
+				l.Error(lua.LString("wrong number or type of arguments"), 1)
+				return 0
+			}
 			res := &lua.LTable{}
 			res.RawSetString("err", lua.LString(msg))
 			l.Push(res)
 			return 1
 		},
 		"status_reply": func(l *lua.LState) int {
-			msg := l.CheckString(1)
+			v := l.Get(1)
+			msg, ok := v.(lua.LString)
+			if !ok {
+				l.Error(lua.LString("wrong number or type of arguments"), 1)
+				return 0
+			}
 			res := &lua.LTable{}
 			res.RawSetString("ok", lua.LString(msg))
 			l.Push(res)

